File: Backend/src/Api.Domain/Models/CarrosModel.cs
using System;

namespace Api.Domain.Models
{
    public class CarrosModel : BaseModel
    {
        private string _nome_carro;
        public string Nome_Carro
        {
            get { return _nome_carro; }
            set { _nome_carro = value; }
        }

        private Guid _TipoCarroId;
        public Guid TipoCarroId
        {
            get { return _TipoCarroId; }
            set { TipoCarroId = value; }
        }

        private Guid _CoresCarroId;
        public Guid CoresCarroId
        {
            get { return _CoresCarroId; }
            set { CoresCarroId = value; }
        }

    }
}

File: Frontend/SysCar/src/app/models/carros/carros.model.ts
import { BaseModel } from "../base.model";

export class CarrosModel extends BaseModel {

    NomeCarro: string
    CoresCarroId: string
    TiposCarroId: string

    constructor(
        NomeCarro: string,
        CoresCarroId: string,
        TiposCarroId: string,

    ) { super(); }

    // tslint:disable-next-line: member-ordering
    public static fromJson(jsonData: any): CarrosModel {
        return Object.assign(c => new CarrosModel(c.NomeCarro, c.CoresCarroId, c.TiposCarroId), jsonData);
    }
}
